src/Compilers/Server/ServerShared/DiagnosticListener.cs

```diff
@@ -16,10 +16,15 @@ internal interface IDiagnosticListener
         void UpdateKeepAlive(TimeSpan timeSpan);
 
         /// <summary>
+        /// Called when a connection to the server occurs.
+        /// </summary>
+        void Connection();
```

Comment:
Could we call it `ReceivedConnection` or similar? Just `Connection` is pretty obtuse.